sdk/python/kfp/dsl/for_loop_test.py

```diff
@@ -136,6 +136,23 @@ def test_loop_artifact_argument_from_pipeline_channel(
         self.assertEqual(loop_argument.items_or_pipeline_channel, channel),
         self.assertEqual(str(loop_argument), expected_serialization_value)
 
+    @parameterized.parameters(
+        {
+            'channel':
+                pipeline_channel.PipelineArtifactChannel(
+                    name='param1',
+                    channel_type='system.Artifact@0.0.1',
+                    task_name='task1',
+                    is_artifact_list=False,
+                ),
+        },
+    )
+    def test_loop_artifact_argument_from_single_pipeline_channel_raises_error(
+            self, channel):
+        self.assertRaises(ValueError,
```

Comment:
Can we use `assertRaisesRegex` to ensure we're catching the correct `ValueError` and that, throughout future changes, we don't "lose" this `ValueError` to a different downstream `ValueError`?

This is the [prevailing pattern](https://github.com/kubeflow/pipelines/blob/b6247fb8e489bb3dd8495fc6b4e9a181755c8896/sdk/python/kfp/dsl/for_loop_test.py#L194) in the SDK source.